pkg/controller/azvolume.go

```diff
@@ -334,7 +334,8 @@ func (r *ReconcileAzVolume) triggerDelete(ctx context.Context, azVolume *azdiskv
 
 					deleteErr = r.deleteVolume(cloudCtx, azVolume)
 				}
-				if deleteErr != nil {
+				// if volume deletion failed and the driver is uninstalling, delete azvolume CRI to let uninstallation proceed
+				if deleteErr != nil && !r.isDriverUninstall() {
```

Comment:
I am not sure if we should be handling this here. I think we should update the error first and handle driver uninstall case on the next reconciler iteration on the update event this will be triggering.